api/guid/views.py

```diff
@@ -0,0 +1,39 @@
+from django import http
+from rest_framework.exceptions import NotFound
+from rest_framework import permissions as drf_permissions
+
+from framework.guid.model import Guid
+from framework.auth.oauth_scopes import CoreScopes, ComposedScopes
+from api.base.exceptions import NotImplementedError
+from api.base import permissions as base_permissions
+from api.base.views import JSONAPIBaseView
+
+
+class GuidRedirect(JSONAPIBaseView):
+
```

Comment:
We should do a quick bit of documentation on this endpoint for the API docs. Maybe:

    """Find an item by its guid.

    This endpoint will redirect you to the most appropriate URL given its OSF GUID. If you give it a GUID for a node,
    you will be redirected to /v2/nodes/{node-id}, for example, while giving it a user's GUID will redirect you to
    /v2/users/{user-id}. If the GUID does not resolve, you will receive a 410 GONE response.

    #This Request/Response

    """